File: backend/greennest/src/main/java/com/greennest/backend/service/OrderService.java
package com.greennest.backend.service;

import com.greennest.backend.dto.request.OrderCreateRequest;
import com.greennest.backend.dto.response.OrderResponse;
import com.greennest.backend.entity.Order;
import com.greennest.backend.entity.User;

import java.util.List;

public interface OrderService {
    
    OrderResponse createOrder(User user, OrderCreateRequest request);
    List<OrderResponse> getUserOrders(User user);
    OrderResponse getOrderById(Long orderId, User user);
    OrderResponse updateOrderStatus(Long orderId, Order.OrderStatus status);
    List<OrderResponse> getAllOrders();
    String generateOrderNumber();
}

File: greennest-frontend/src/services/orderService.js
import api from './api';

export const orderService = {
  createOrder: async (orderData) => {
    const response = await api.post('/orders/create', orderData);
    return response.data;
  },

  getUserOrders: async () => {
    const response = await api.get('/orders');
    return response.data;
  },

  getOrderById: async (orderId) => {
    const response = await api.get(`/orders/${orderId}`);
    return response.data;
  }
};
